spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,99 @@
+require "rails_helper"
+
+RSpec.describe UsersController, type: :controller do
+  include SessionHelper
+  let!(:user) { FactoryBot.create(:user) }
+  let(:invalid_email) { FactoryBot.attributes_for(:user, email: "bjsadbjkasd") }
```

Comment:
`let` và `let!` khác nhau thế nào em ơi

Reply:
Dạ let! nó sẽ ghi nhớ lại kết quả của một example và nó sẽ có thể được khai báo ở nhiều khối it, nó được gọi trong một khối before ngầm, do đó kết quả đã được tính toán và cache lại trước khi khổi it được thực thi. Còn let thì nó chỉ thực thi khi mà được gọi đến, let và let! em thấy tương tự như một cái là biến local và một cái là biến instance ạ